### ISSUE TITLE
metric collector - HubResourceID required even if UploadTarget is IoTMessage

### ISSUE THREAD

**Issue by Banchio** [Azure/iotedge#4905]
## Expected Behavior
In the docs https://labs.iotedge.dev/codelabs/edgemon-preview/#4 states that HubResourceID is not required if UploadTarget is IoTMessage

## Current Behavior
Module fails to start if UploadTarget is IoTMessage and HubResourceID is not specified

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Use Set Modules on an Edge 1.2
2. Use azureiotedge/azureiotedge-metrics-collector:beta1 as image name
3. Set environment variable UploadTarget to IoTMessage
4. Commit

## Context (Environment)
edge runtime 1.2 on VM Ubuntu 18.04

**Comment by and-rewsmith** [comment]
@Banchio You are correct and that there is a mismatch between the docs and the logic for required parameters. I will open a bug. Thanks for catching this!

I did want to point out though that this is not the exception I would expect to see in this case. This stack trace indicates a timeout when sending a message, however in the case you describe I wouldn't expect it to get this far. I would expect it to fail at the very beginning.
